src/commands/link.rs

```diff
@@ -143,176 +129,84 @@ fn select_environment(
+            prompt_team_projects(projects)?
         }
     });
     Ok(project)
 }
 
-fn select_team(
+fn select_workspace(
     project: Option<String>,
-    team: Option<String>,
-    me: &queries::user_projects::UserProjectsMe,
-) -> Result<Team<'_>, anyhow::Error> {
-    let uuid_regex =
-        Regex::new(r#"(?i)^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}"#)
-            .unwrap();
-    let team = match (project.as_ref(), team.as_ref()) {
-        (Some(project), None) if uuid_regex.is_match(project) => {
-            // It's a project id, figure out team
-            if me
-                .projects
-                .edges
-                .iter()
-                .any(|pro| pro.node.id.to_lowercase() == project.to_lowercase())
-            {
-                fake_select("Select a team", "Personal");
-                Team::Personal
-            } else if let Some(team) = me.teams.edges.iter().find(|team| {
-                team.node
-                    .projects
-                    .edges
-                    .iter()
-                    .any(|proj| proj.node.id.to_lowercase() == project.to_lowercase())
-            }) {
-                fake_select("Select a team", &team.node.name);
-                Team::Team(&team.node)
-            } else {
-                return Err(RailwayError::ProjectNotFound.into());
-            }
-        }
+    workspace_name: Option<String>,
+    workspaces: Vec<Workspace>,
+) -> Result<Workspace, anyhow::Error> {
+    let workspace = match (project, workspace_name) {
         (Some(project), None) => {
-            // this means project name without team
-            if me.teams.edges.is_empty() {
-                // no teams, so it's personal
-                // if there is a project that has the same name
-                if me
-                    .projects
-                    .edges
-                    .iter()
-                    .any(|p| p.node.name.to_lowercase() == project.to_lowercase())
-                {
-                    fake_select("Select a team", "Personal");
-                    Team::Personal
-                } else {
-                    return Err(RailwayError::ProjectNotFound.into());
-                }
+            // It's a project id or name, figure out workspace
+            if let Some(workspace) = workspaces.iter().find(|w| {
+                w.projects().iter().any(|pro| {
+                    pro.id().to_lowercase() == project.to_lowercase()
+                        || pro.name().to_lowercase() == project.to_lowercase()
+                })
+            }) {
+                fake_select("Select a workspace", workspace.name());
+                workspace.clone()
             } else {
-                prompt_teams(me)?
+                prompt_workspaces(workspaces)?
             }
         }
         (None, Some(team_arg)) | (Some(_), Some(team_arg)) => {
-            match team_arg.to_lowercase().as_str() {
-                "personal" => {
```

Comment:
Wondering what to do about this. It's technically a regression removing the option to set the personal workspace like that. But it's not a thing anymore, so we wont be able to infer a personal workspace pretty soon.